src/HonzaBotner.Discord.Services/Commands/Polls/AbcPoll.cs

```diff
@@ -35,4 +40,55 @@ public AbcPoll(string authorMention, string question, List<string> options)
     public AbcPoll(DiscordMessage message) : base(message)
     {
     }
+
+    public async Task AddOptionsAsync(DiscordClient client, IEnumerable<string> newOptions)
+    {
+        if (ExistingPollMessage is null)
+        {
+            throw new InvalidOperationException("You can edit only poll constructed from sent message.");
+        }
+
+        NewChoices = newOptions.ToList();
+
+        if (NewChoices.Count + ExistingPollMessage.Embeds[0].Fields.Count > OptionsEmoji.Count)
+        {
+            throw new PollException($"Too many options. Maximum options is {OptionsEmoji.Count}.");
+        }
+
+        List<string> emojisToAdd = OptionsEmoji;
+        emojisToAdd.RemoveAll(emoji =>
+            ExistingPollMessage.Reactions.Select(rect => rect.Emoji).Contains(DiscordEmoji.FromName(client, emoji)));
+
+        emojisToAdd = emojisToAdd.GetRange(
+            0, Math.Min(
+                Math.Min(
+                    emojisToAdd.Count,
+                    20 - (ExistingPollMessage.Reactions.Count <= 20 ? ExistingPollMessage.Reactions.Count : 20)),
```

Comment:
You have put `Math.Min` everywhere, so why not here?

Reply:
There would be too many Math.Mins 😁😅